shell/platform/windows/window_state.h

```diff
@@ -0,0 +1,73 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_WINDOWS_FLUTTER_WINDOW_STATE_H_
+#define FLUTTER_SHELL_PLATFORM_WINDOWS_FLUTTER_WINDOW_STATE_H_
+
+#include "flutter/shell/platform/common/cpp/client_wrapper/include/flutter/plugin_registrar.h"
+#include "flutter/shell/platform/common/cpp/incoming_message_dispatcher.h"
+#include "flutter/shell/platform/embedder/embedder.h"
+#include "flutter/shell/platform/windows/key_event_handler.h"
+#include "flutter/shell/platform/windows/keyboard_hook_handler.h"
+#include "flutter/shell/platform/windows/platform_handler.h"
+#include "flutter/shell/platform/windows/text_input_plugin.h"
+
+struct flutter::FlutterWindow;
+
+// Struct for storing state within an instance of the windows native (HWND or
+// CoreWindow) Window.
+struct FlutterDesktopWindowControllerState {
+  //// The win32 window that owns this state object.
+  std::unique_ptr<flutter::FlutterWindow> window;
+
+  // The handle to the Flutter engine instance.
+  FlutterEngine engine;
+
+  // The window handle given to API clients.
+  std::unique_ptr<FlutterDesktopWindow> window_wrapper;
+};
+
+// Opaque reference for the native windows itself. This is separate from the
+// controller so that it can be provided to plugins without giving them access
+// to all of the controller-based functionality.
+struct FlutterDesktopWindow {
+  // The GLFW window that (indirectly) owns this state object.
+  flutter::FlutterWindow* window;
+
+  // Whether or not to track mouse movements to send kHover events.
+  bool hover_tracking_enabled = true;
```

Comment:
It seems like some of these struct members are unused, in which case they shouldn't be in the code.

Reply:
It would still be good to prune anything that's flat out unused though.